router/internal/persistedoperation/operationstorage/redis/rdcloser.go

```diff
@@ -60,6 +71,33 @@ func NewRedisCloser(opts *RedisCloserOptions) (RDCloser, error) {
 	return rdb, nil
 }
 
+func addClusterUrlsToQuery(opts *RedisCloserOptions, parsedUrl *url.URL) {
+	if len(opts.URLs) <= 1 {
+		return
+	}
+
+	queryVals := parsedUrl.Query()
+	for _, rawURL := range opts.URLs[1:] {
+		secondaryURL, parseErr := url.Parse(rawURL)
+		if parseErr != nil {
+			opts.Logger.Warn(fmt.Sprintf("Skipping invalid Redis URL %q: %v", rawURL, parseErr))
+			continue
+		}
+
+		// Strip schema, username, and password
+		addr := secondaryURL.Host
+		if secondaryURL.User != nil && parsedUrl.User != nil && parsedUrl.User.String() != "" && secondaryURL.User.Username() != parsedUrl.User.Username() {
+			opts.Logger.Warn(fmt.Sprintf("Stripping credentials from secondary Redis address: %q", addr))
```

Comment:
nitpick, zap is a structured logging library, you can write something like this instead of using fmt.Sprintf which essentially turns Zap.Logger into Zap.SugaredLogger performance wise.
```go
opts.Logger.Warn("Stripping credentials from secondary Redis address", zap.String("addr", addr))
```

If you'd really like to use Printf semantics, Zap supports this via turning the logger into a sugaredlogger and vise versa at any point, like this:
```go
sl := l.Sugar()

sl.Warnf(...)

l = sl.Desugar()